test/gtest/core/test_bcast.cc

```diff
@@ -54,10 +54,13 @@ class test_bcast : public UccCollArgs, public ucc::test
     }
     void data_fini(UccCollCtxVec ctxs)
     {
-        for (gtest_ucc_coll_ctx_t* ctx : ctxs) {
+        for (auto r = 0; r < ctxs.size(); r++) {
+            gtest_ucc_coll_ctx_t *ctx = ctxs[r];
             ucc_coll_args_t* coll = ctx->args;
             UCC_CHECK(ucc_mc_free(coll->src.info.buffer, mem_type));
-            UCC_CHECK(ucc_mc_free(ctx->init_buf, UCC_MEMORY_TYPE_HOST));
+            if (r == coll->root) {
+                UCC_CHECK(ucc_mc_free(ctx->init_buf, UCC_MEMORY_TYPE_HOST));
```

Comment:
If we are to fix, then lets do it right: the init_buf array is allocated always on host (UCC_MEMORY_TYPE_HOST is provided). We should not use ucc_mc_alloc in this case, but rather use ucc_malloc (and hence ucc_free).

Reply:
in another PR